### DISCUSSION TITLE
Idea - replace floats with posits in Mojo

### DISCUSSION THREAD

**Discussion by rbf22** [modular/modular#2684] [Ideas]
There seem to be a lot of benefits to using posits to represent reals rather than floats. There is also nice efficiency that comes from some complex operations like inverses and exponents. I am not an expert in these ideas, but here are a few articles/posts that have discussed this in detail. Maybe it is the case that Mojo is built from the ground up with posits, and I don't know. But I wanted to raise this here so that others can share thoughts or comments. 

1. https://spectrum.ieee.org/floating-point-numbers-posits-processor
2. https://www.cs.cornell.edu/courses/cs6120/2019fa/blog/posits/
3. https://stackoverflow.com/questions/63651414/why-aren-t-posit-arithmetic-representations-commonly-used
4. https://posithub.org/docs/Posits4.pdf

**Comment by nmsmith** [comment]
Mojo doesn't have any built-in data types. The Mojo community is free to add any numeric types that it wants.

Supporting posits won't be very useful unless CPUs and GPUs have instructions for operating on them. Currently, no major vendor supports them.
